stock_picking_production_lot_manual_selector/models/stock_move_line.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    use_manual_lots = fields.Boolean(
+        related="picking_id.use_manual_lots",
+    )
+
+    manual_production_lot_id = fields.Many2one(
+        "stock.production.lot",
+        "Lot/Serial Number (manual)",
+        domain="[('product_id', '=', product_id), ('company_id', '=', company_id)]",
+        check_company=True,
+        copy=False,
+        help="This field is used to manually select a production lot.",
+    )
+
+    @api.onchange("manual_production_lot_id")
+    def onchange_manual_production_lot_id(self):
+        if self.manual_production_lot_id:
+            self.lot_id = self.manual_production_lot_id
+
+    def _action_done(self):
+        lines_to_block = self.filtered(
+            lambda line: line.use_manual_lots and not line.manual_production_lot_id
```

Comment:
I presumed the check should be done on the lot type of the product to keep standard behavior in that case:
`lambda line: line.use_manual_lots and not line.manual_production_lot_id and line.product_id.tracking != 'none')`

And IMO this is better because it keeps the quantity check later in the workflow:
`You cannot validate a transfer if no quantities are reserved nor done. To force the transfer, switch in edit mode and encode the done quantities.`

Reply:
Adding both tracking type AND qty done